### PR TITLE
Jose/refactor

### PR DESCRIPTION
**Pull request by jotathebest**
Refactor to add support by default to industrial.api.ubidots.com. Several code pieces were made again to increase performance and code style. Added support for device types.

### CODE REVIEW

**Comment by csaraivarocha** [comment]
It doesn’t work with ESP32, does it?

File: library.propierties [hunk 1]
```diff
@@ -1,7 +1,7 @@
-name=Ubidots Arduino Ethernet
-version=2.0.1
-author=María Carlina Hernandez <maria@ubidots.com>, Mateo Velez
-maintainer=María Carlina Hernandez <maria@ubidots.com>, Jose Garcia <jose.reyes@ubidots.com>
+name=Ubidots ESP8266
+version=1.4
```
Comment (woakas) [inline library.propierties]: change this to version 2.1.0

File: library.propierties [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
+author=José García <jose.garcia@ubidots.com>, María Carlina Hernandez <maria@ubidots.com>, Mateo Velez
```
Comment (woakas) [inline library.propierties]: Remove Mateo Velez author.

File: UbidotsMicroESP8266.h [hunk 3]
```diff
@@ -32,35 +32,17 @@ Modified by: Maria Carlina Hernandez
...
-#define DEFAULT_DEVICE_NAME "ESP8266"
-#endif
-
-const float ERROR_VALUE = -3.4028235E+8;
+namespace {
+  const char * DEFAULT_DEVICE_NAME = "ESP8266";
+  const char * UBIDOTS_HTTP_SERVER = "industrial.api.ubidots.com";
+  const char * UBIDOTS_TCP_SERVER = "industrial.api.ubidots.com";
+  const char * USER_AGENT = "UbidotsESP8266";
+  const char * VERSION = "1.4";
```
Comment (woakas) [inline UbidotsMicroESP8266.h]: This should be the same library version: 2.1.0

File: Examples/UbidotsDataLogger/UbidotsDataLogger.ino [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  End of HTTP Block Example
+  **********************************/
+
+  /* ------------------------------------ */
+
+  /**********************************
+  Uncomment this block if you wish to
+  send data through TCP
+  **********************************/
+  //client.setServerEndpoint("translate.ubidots.com");  // Uncomment this line if you are an educational user
```
Comment (woakas) [inline Examples/UbidotsDataLogger/UbidotsDataLogger.ino]: Move this line to setup

File: Examples/UbidotsSaveValueWithLabel/UbidotsSaveValueWithLabel.ino [hunk 5]
```diff
@@ -9,14 +9,19 @@ Ubidots client(TOKEN);
 void setup(){
     Serial.begin(115200);
     client.wifiConnection(WIFISSID, PASSWORD);
-    //client.setDebug(true); // Uncomment this line to set DEBUG on
+    //client.setServerEndpoint("translate.ubidots.com");  // Uncomment this line if you are an educational user
```
Comment (woakas) [inline Examples/UbidotsSaveValueWithLabel/UbidotsSaveValueWithLabel.ino]: Change this line to things.ubidots.com.

File: Readme.md [hunk 6]
```diff
@@ -46,308 +45,87 @@ TXD | RXD | TX
...
-    client.add("temperature", value1);
-    //client.add("switch", value2);
-    client.sendAll(true);
-    delay(5000);
-
-}
+Ubidots(char* token)
+Ubidots(char* token, const char * server)
 ```
+> Creates an Ubidots instance, you must setup as input your Ubidots TOKEN, the server is optional and is set by default to "industrial.api.ubidots.com" for sending data using an Industrial account. You can set another endpoint using the `setServerEndpoint()` method.
```
Comment (woakas) [inline Readme.md]: Add a reference to Ubidots for education.

File: UbidotsMicroESP8266.cpp [hunk 7]
```diff
@@ -83,7 +79,7 @@ long Ubidots::getVarTimestamp(char* id) {
   sprintf(data, "%sHost: things.ubidots.com\r\nUser-Agent: %s/%s\r\n", data, USER_AGENT, VERSION);
```
Comment (woakas) [inline UbidotsMicroESP8266.cpp]: Change this line with a constant.

File: UbidotsMicroESP8266.cpp [hunk 8]
```diff
@@ -158,7 +154,7 @@ char* Ubidots::getVarContext(char* id) {
   sprintf(data, "%sHost: things.ubidots.com\r\nUser-Agent:%s/%s\r\n", data, USER_AGENT, VERSION);
```
Comment (woakas) [inline UbidotsMicroESP8266.cpp]: Change this line to use a constant

**Comment by jotathebest** [comment]
@csaraivarocha Greetings, this library has been tested using an ESP8266, any compatibility with ESP32 is not warranted.

All the best

**Pull request closed without merging**